File: VECTO/Input Files/MockComponents.vb
Imports TUGraz.VectoCommon.InputData

Public Class MockComponents
    Implements IVehicleComponentsDeclaration

    Public Property AirdragInputData As IAirdragDeclarationInputData _
        Implements IVehicleComponentsDeclaration.AirdragInputData

    Public Property GearboxInputData As IGearboxDeclarationInputData _
        Implements IVehicleComponentsDeclaration.GearboxInputData

    Public Property TorqueConverterInputData As ITorqueConverterDeclarationInputData _
        Implements IVehicleComponentsDeclaration.TorqueConverterInputData

    Public Property AxleGearInputData As IAxleGearInputData Implements IVehicleComponentsDeclaration.AxleGearInputData

    Public Property AngledriveInputData As IAngledriveInputData _
        Implements IVehicleComponentsDeclaration.AngledriveInputData

    Public Property EngineInputData As IEngineDeclarationInputData _
        Implements IVehicleComponentsDeclaration.EngineInputData

    Public Property AuxiliaryInputData As IAuxiliariesDeclarationInputData _
        Implements IVehicleComponentsDeclaration.AuxiliaryInputData

    Public Property RetarderInputData As IRetarderInputData Implements IVehicleComponentsDeclaration.RetarderInputData

    Public Property PTOTransmissionInputData As IPTOTransmissionInputData _
        Implements IVehicleComponentsDeclaration.PTOTransmissionInputData

    Public Property AxleWheels As IAxlesDeclarationInputData Implements IVehicleComponentsDeclaration.AxleWheels
    Public ReadOnly Property BusAuxiliaries As IBusAuxiliariesDeclarationData Implements IVehicleComponentsDeclaration.BusAuxiliaries
    Public ReadOnly Property ElectricStorage As IElectricStorageSystemDeclarationInputData Implements IVehicleComponentsDeclaration.ElectricStorage
    Public ReadOnly Property ElectricMachines As IElectricMachinesDeclarationInputData Implements IVehicleComponentsDeclaration.ElectricMachines
    Public ReadOnly Property IEPC As IIEPCDeclarationInputData Implements IVehicleComponentsDeclaration.IEPC
    Public ReadOnly Property FuelCellSystem As IFuelCellSystemDeclarationInputData Implements IVehicleComponentsDeclaration.FuelCellSystem
    Public ReadOnly Property AxlePowertrainInputData As IList(Of IAxlePowertrainDeclarationInputData) Implements IVehicleComponentsDeclaration.AxlePowertrainInputData
    Public ReadOnly Property Generator As ElectricMachineEntry(Of IElectricMotorDeclarationInputData) Implements IVehicleComponentsDeclaration.Generator

End Class

File: VectoCore/VectoCoreTest/Utils/MockComponents.cs
using System.Collections.Generic;
using TUGraz.VectoCommon.InputData;

namespace TUGraz.VectoCore.Tests.Utils
{
	public class MockComponents : IVehicleComponentsDeclaration
	{
		public IAirdragDeclarationInputData AirdragInputData { get; set; }

		public IGearboxDeclarationInputData GearboxInputData { get; set; }

		public ITorqueConverterDeclarationInputData TorqueConverterInputData { get; set; }

		public IAxleGearInputData AxleGearInputData { get; set; }

		public IAngledriveInputData AngledriveInputData { get; set; }

		public IEngineDeclarationInputData EngineInputData { get; set; }

		public IAuxiliariesDeclarationInputData AuxiliaryInputData { get; set; }

		public IRetarderInputData RetarderInputData { get; set; }

		public IPTOTransmissionInputData PTOTransmissionInputData { get; set; }

		public IAxlesDeclarationInputData AxleWheels { get; set; }

		public IBusAuxiliariesDeclarationData BusAuxiliaries { get; }

		public IElectricStorageSystemDeclarationInputData ElectricStorage { get; }

		public IElectricMachinesDeclarationInputData ElectricMachines { get; }

		public IIEPCDeclarationInputData IEPC { get; }

		public IFuelCellSystemDeclarationInputData FuelCellSystem { get; }

        public IList<IAxlePowertrainDeclarationInputData> AxlePowertrainInputData { get; }

        public ElectricMachineEntry<IElectricMotorDeclarationInputData> Generator { get; }
    }
}